fuel-core-interfaces/src/model/block.rs

```diff
@@ -74,14 +84,25 @@ impl FuelBlockHeader {
             self.hash()
         }
     }
+
+    pub fn transactions_root(txs: &[Transaction]) -> Bytes32 {
+        let mut storage = StorageMap::new();
+        let mut tree = MerkleTree::new(&mut storage);
+        for tx in txs {
+            // serialize tx into canonical format for hashing
+            let ser_tx = tx.clone().to_bytes();
+            tree.push(&ser_tx).expect("infallible storage");
+        }
+        tree.root().expect("infallible storage").into()
```

Comment:
You can instead use the in-memory version of the Merkle tree to provide in-memory, infallible storage for calculating the transactions root. This would look something like:

```
use fuel_merkle::binary::in_memory::MerkleTree;
...

let merkle_tree = MerkleTree::new();
for tx in txs {
    // serialize tx into canonical format for hashing
    let ser_tx = tx.clone().to_bytes();
    tree.push(&ser_tx);
}
tree.root().into()
```